app/assets/javascripts/discourse/initializers/copy-codeblocks.js

```diff
@@ -0,0 +1,146 @@
+  const range = window.document.createRange();
+  selection.removeAllRanges();
+  range.selectNode(span);
+  selection.addRange(range);
+
+  // Copy text to the clipboard
+  let success = false;
+  try {
+    success = window.document.execCommand("copy");
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.log("error", err);
+  }
+
+  // Cleanup
+  selection.removeAllRanges();
+  window.document.body.removeChild(span);
+
+  return success
+    ? Promise.resolve()
+    : Promise.reject(
+        new DOMException("The request is not allowed", "NotAllowedError")
+      );
+}
+
+let _copyCodeblocksClickHandlers = {};
+let _fadeCopyCodeblocksRunners = {};
+
+export default {
+  name: "copy-codeblocks",
+
+  initialize(container) {
+    withPluginApi("0.8.7", api => {
+      function _cleanUp() {
+        Object.values(_copyCodeblocksClickHandlers || {}).forEach(handler =>
+          handler.removeEventListener("click", _handleClick)
+        );
+
+        Object.values(_fadeCopyCodeblocksRunners || {}).forEach(runner =>
+          cancel(runner)
+        );
+
+        _copyCodeblocksClickHandlers = {};
+        _fadeCopyCodeblocksRunners = {};
+      }
+
+      function _handleClick(event) {
+        if (!event.target.classList.contains("copy-cmd")) {
+          return;
+        }
+
+        const button = event.target;
+        const code = button.nextSibling;
+
+        if (code) {
+          clipboardCopy(code.innerText.trim()).then(() => {
+            button.classList.add("copied");
+
+            const commandId = button.dataset.commandId;
+
+            if (_fadeCopyCodeblocksRunners[commandId]) {
+              cancel(_fadeCopyCodeblocksRunners[commandId]);
+              delete _fadeCopyCodeblocksRunners[commandId];
+            }
+
+            _fadeCopyCodeblocksRunners[commandId] = later(
+              () => button.classList.remove("copied"),
+              3000
+            );
+          });
+        }
+      }
+
+      function _attachCommands(postElements, helper) {
+        if (!helper) {
+          return;
+        }
+
+        const siteSettings = container.lookup("site-settings:main");
+        const { isIE11 } = container.lookup("capabilities:main");
+        if (!siteSettings.show_copy_button_on_codeblocks || isIE11) {
+          return;
+        }
+
+        const commands = postElements[0].querySelectorAll(
+          ":scope > pre > code"
+        );
+
+        const post = helper.getModel();
+
+        if (!commands.length || !post) {
+          return;
+        }
+
+        const postElement = postElements[0];
+        _copyCodeblocksClickHandlers[post.id] = postElement;
+
+        commands.forEach((command, commandIndex) => {
+          const button = document.createElement("button");
+          button.classList.add("btn", "nohighlight", "copy-cmd");
+          button.innerHTML = iconHTML("copy");
+          button.dataset.commandId = `${post.id}-${commandIndex}`;
```

Comment:
I hope that in the case when a post is edited to add/remove a comand we call `cleanUp` before decorating again because these would stamp on the other `id`s. Something to test!

Reply:
OFC we don't :p good catch, will think about it...

Reply:
Most recent commit (patch from @jjaffeux) should address this.